app/controllers/api_controller.rb

```diff
@@ -28,6 +30,20 @@ def user_for_paper_trail
 
 private
 
+  # NB: for new controllers
+  def validate_required_idempotency_key
+    idempotence_validator.call(request.headers['IDEMPOTENCY_KEY'], required: true)
+  end
+
+  # NB: for legacy controllers
+  def validate_optional_idempotency_key
```

Comment:
Is there a future plan to move all controllers to use the `idempotency_key`, or just the state endpoints?

Reply:
Ultimately, I think all end points should expect an idempotency key. Enforcing that now however will break everything, hence the idea of an optional one.